src/widget.js

```diff
@@ -214,6 +219,31 @@ export default class Widget extends Plugin {
 	}
 
 	/**
+	 * Extends the {@link module:engine/model/selection~Selection document's selection} to span the entire
+	 * content of the nested editable if already anchored in one.
+	 *
+	 * See: {@link module:engine/model/schema~Schema#getLimitElement}.
+	 *
+	 * @private
+	 */
+	_selectAllNestedEditableContent() {
+		const modelDocument = this.editor.document;
+		const modelSelection = modelDocument.selection;
+		const schema = modelDocument.schema;
+		const limitElement = schema.getLimitElement( modelSelection );
+
+		if ( modelSelection.getFirstRange().root == limitElement ) {
+			return false;
+		}
+
+		modelDocument.enqueueChanges( () => {
+			modelSelection.setIn( limitElement );
```

Comment:
Hum... hum... That's good for now (when we have a figcaption with only text inside), but that's definitely not good for the future. Selection cannot be placed outside paragraphs for instance. So we need something like `Selection#selectEntireContent( element )` which will use the schema to query where it can be located. However, since this will require the schema I wonder if this method shouldn't end in some other place – `DataController` maybe?

cc @pjasiun